### DISCUSSION TITLE
Best Practice for Large Documents

### DISCUSSION THREAD

**Discussion by morankyle** [explosion/spaCy#8561] [Help: Best practices]
Currently have a Spacy pipeline which includes various built-in pipes such as ner, parser, etc, as well a custom component for extracting SVOs and SpacyTextBlob. 

The model runs into issues with very large documents, I've read Spacy needs 1GB per 100,000. Is there any best practice for splitting up documents before sending them thru the pipeline in? Or is it possible to somehow split the doc into spans and send those thru the pipeline individually?

**Comment by polm** [comment]
If you're running into memory errors due to large individual docs, usually you should be able to split them into paragraphs without a loss in performance. 

This is mentioned [in the same comment](https://github.com/explosion/spaCy/issues/7268#issuecomment-828280225) that suggests 1GB per 100,000 characters, actually.

**Reply by morankyle** [reply]
Awesome thank you! Actually came across that comment in a different Stack Over Flow thread so that was helpful!
